tests/unit/tf/inputs/test_embedding.py

```diff
@@ -214,6 +214,50 @@ def test_from_pretrained(self, trainable, music_streaming_data: Dataset):
         else:
             np.testing.assert_array_almost_equal(weights, embedding_table.table.embeddings)
 
+    @pytest.mark.parametrize("trainable", [True, False])
+    def test_pretrained_from_InputBlockV2(self, trainable, music_streaming_data: Dataset):
+        vocab_size = music_streaming_data.schema.column_schemas["item_id"].int_domain.max + 1
+        embedding_dim = 32
+        weights = np.random.rand(vocab_size, embedding_dim)
+        pre_trained_weights_df = pd.DataFrame(weights)
+
+        embed_dims = {}
+        embed_dims["item_id"] = pre_trained_weights_df.shape[1]
+        embeddings_init = {
+            "item_id": mm.TensorInitializer(weights),
+        }
+
+        embeddings_block = mm.Embeddings(
+            music_streaming_data.schema,
+            infer_embedding_sizes=True,
+            embeddings_initializers=embeddings_init,
+            trainable={"item_id": trainable},
+            embedding_dims=embed_dims,
+        )
+        input_block = mm.InputBlockV2(music_streaming_data.schema, embeddings=embeddings_block)
+
+        model = mm.DCNModel(
+            music_streaming_data.schema,
+            depth=2,
+            input_block=input_block,
+            deep_block=mm.MLPBlock([64, 32]),
+            prediction_tasks=mm.BinaryClassificationTask("click"),
+        )
+        model_test(model, music_streaming_data)
+
+        if trainable:
+            np.testing.assert_raises(
+                AssertionError,
+                np.testing.assert_array_almost_equal,
+                weights,
+                model.layers[0].layers[0].layers[1]["embeddings"]["item_id"].table.embeddings,
```

Comment:
wondering if there is a way to find a layer in a model that is not tied to the depth in the structure

Reply:
good question. idk, there might be.

Reply:
try replacing `model.layers[0].layers[0].layers[1]["embeddings"]` with `embeddings_block`